### ISSUE TITLE
Use additional table fields

### ISSUE THREAD

**Issue by yougotnet** [spatie/laravel-activitylog#462]
I need another field in the table customer_id that is the customer of the causer_id.  In my application, all users belong to a customer and it would be faster to query a column customer_id for the customer level and causer_id for the user level.

Is this possible?

**Comment by Gummibeer** [comment]
My first idea would be an observer for the `Activity` model which adds the `customer_id` to the `custom_properties`.
https://laravel.com/docs/5.7/eloquent#observers

**Comment by yougotnet** [comment]
It probably could, but that requires 3 lines of code; if it was built into the package, it could be incorporated into 1 line of code.

**Comment by freekmurze** [comment]
@Gummibeer I don't think we'd need to support that callback as a second argument. You can do this:

```
tap(activity()->log('test'), $callback) {
   // do stuff here...
}
```

**Comment by yougotnet** [comment]
That seems you are saving the log first and then after it is saved, then do more stuff.  Just seems like adding a band aid and not really addressing the issue.

It is common to want to add additional fields into the activity log; I'm not seeing the difficulty in adding it to the tool so that it can be more streamlined and efficient.

**Comment by Gummibeer** [comment]
@freekmurze like @yougotnet said: this will save the model twice - something we should prevent for logs. For me the table/model shouldn't even allow an update, but this is something else.

**Comment by freekmurze** [comment]
Ok, thanks for clarifying. Instead of adding a second parameter to `log` we could introduce a new method right?

activity()
->with(function(Activity $activity) { $activity->customer_id = $activity->causer->customer->getKey(); })
->log('message');

That `with` name could be improved. Maybe `tap`? 😄

**Comment by yougotnet** [comment]
Sounds great! That would allow any number of additional fields for the model.

**Comment by cdoouteiro** [comment]
I really need this too to store the causer IP (and maybe other fields in the future). Does anyone have a working code of doing this with observer ? i tried saved/created/saving/creating methods on observer class with no success.

**Comment by Gummibeer** [comment]
I've added the discussed `tap()` method in #472 - everyone is welcome to test it and/or give a review.

**Comment by Gummibeer** [comment]
This was released as `v3.2.0`
https://github.com/spatie/laravel-activitylog/blob/master/CHANGELOG.md#320---2019-01-29
